### PR TITLE
ifndef BYO_CRYPTO

### PR DESCRIPTION
**Pull request by ilevyor**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by justinboswell — CHANGES_REQUESTED** [review]
I don't understand this change, the body_streaming_elg exists whether or not BYO_CRYPTO is enabled, how does this relate?

**Review by TingDaoK — CHANGES_REQUESTED** [review]
it's complaining about on_error not used for `BYO_CRYPTO`, I guess you need to add something like `goto on_error` on line 339. And raise some error

**Comment by ilevyor** [comment]
commit is necessary due to BYO_CRYPTO builds failing in aws-crt-cpp 
https://github.com/awslabs/aws-crt-cpp/runs/3106110215?check_suite_focus=true

> `/root/aws-crt-cpp/crt/aws-c-s3/source/s3_client.c:411:1: error: label 'on_error' defined but not used [-Werror=unused-label]
 on_error:`

File: source/s3_client.c [hunk 1]
```diff
@@ -752,14 +753,6 @@ static struct aws_s3_meta_request *s_s3_client_meta_request_factory_default(
 
     /* Call the appropriate meta-request new function. */
     if (options->type == AWS_S3_META_REQUEST_TYPE_GET_OBJECT) {
-
-        /* If the initial request already has partNumber, the request is not splittable(?). Treat it as a Default
-         * request.
-         * TODO: Still need tests to verify that the request of a part is splittable or not */
-        if (aws_http_headers_has(initial_message_headers, aws_byte_cursor_from_c_str("partNumber"))) {
```
Comment (TingDaoK) [inline source/s3_client.c:759]: why delete this part? And have you tried to use this branch for aws-crt-cpp and see the CI is happy or not?

Comment (ilevyor) [reply source/s3_client.c]: mistake, fixing now. How do I run CI on crt-cpp with s3 set to this branch?

Comment (ilevyor) [reply source/s3_client.c]: BYO_CRYPTO works on crt-cpp with s3 on this branch.
https://github.com/awslabs/aws-crt-cpp/runs/3107882821?check_suite_focus=true

File: source/s3_client.c [hunk 2]
```diff
@@ -337,6 +337,7 @@ struct aws_s3_client *aws_s3_client_new(
         } else {
 #ifdef BYO_CRYPTO
             AWS_ASSERT(false);
+            goto on_error;
```
Comment (TingDaoK) [inline source/s3_client.c:340]: Not sure we want to handle this error.
- If we want to crash in this case, use `AWS_FATAL_ASSERT(false);`
- If we want to handle it more properly, add `aws_raise_error(); /* Some proper error */`

**Review by TingDaoK — APPROVED** [review]

**Review by justinboswell — APPROVED** [review]

**Pull request merged**